tweet_recommendations/other_methods/graph_summarization_for_hashtag_recommendation.py

```diff
@@ -175,53 +177,101 @@ def fit(self, x: pd.DataFrame, y=None, **fit_params) -> "Method":
         self._hashtags_tf_idf_vectorizer = TfidfVectorizer(norm="l2")
         self._hashtags_tf_idf_representation = self._hashtags_tf_idf_vectorizer.fit_transform(document_list)
 
+        if self.verbose:
+            print("Fitting hashtag encoders ...")
+
+        # [:-4] because each hashtag has "_tag" postfix to distinguish it in the graph
+        self._hashtag_encoder.fit([lab[:-4] for lab in self._hashtag_labels])
+
         return self
 
-    def transform(self, x: Union[List[str], str]) -> List[str]:
+    def transform(self, x: Union[Collection[Collection[str]], Collection[str]], **kwargs) -> np.ndarray:
         """
         For a given tweet represented as a list of lemmas recommends hashtags.
         :param x: list of str or str. If list is str, strs are lemmas of the tweet. If single str, it is assumed that
             lemmatization has to be performed.
-        :return: Iterable of recommended hashtags.
+        :key query: Iterable[str]. Optional query hashtag for each tweet in `x`.
+        :return: np.ndarray of recommended hashtags.
         """
-        if isinstance(x, str):
-            x = get_wcrft2_results_for_text(x)
-        if isinstance(x, list):
-            x = ' '.join(x)
+        if isinstance(x[0], str):
+            for i, xi in enumerate(x):
+                x[i] = get_wcrft2_results_for_text(xi)
+        if isinstance(x[0], list):
+            for i, xi in enumerate(x):
+                x[i] = ' '.join(xi)
+
+        query_hashtags = kwargs.get("query", None)
+        if query_hashtags is not None:
+            assert len(query_hashtags) == len(x), \
```

Comment:
Ewentualnie można weryfikować czy query_hashtags są listą a nie pojedynczym stringiem... ale to raczej drobiazg

Reply:
Mogę zrobić, ale rzeczywiście to drobiazg :P